libmscore/scorefile.cpp

```diff
@@ -413,8 +414,36 @@ bool MasterScore::saveFile()
             //
             // step 2
             // remove old backup file if exists
+            // remove the backup file in the same dir as score (the traditional place) if exists
             //
+            QString backupDirString = info.path() + QString(QDir::separator()) + ".backup";
+            if (!dir.cd(".backup")) {
+                  dir.mkdir(".backup");
```

Comment:
what if that mkdir fails?

Reply:
QDir.::mkdir() returns success or failure, so better check that

Reply:
Or blindly do the mkdir(), no check whether it exists prior to that, mkdir() will fail if it does, and (hopefully?!?) succeed if it does not

Reply:
But then the following steps are not needed, resp. would fail

Reply:
So what do you suggest the program should do if mkdir() fails?

Reply:
A `QWarning()` or `QError()`? Then fall back to the old ".score.mscz," maybe?

Reply:
Are you sure about this? Because I don't know what other programs will do if creating a directory fails...

Reply:
I don't know either. But not being able to create a bacupk is less severe than not being able to save the real file, and as far as I can tell this routine has just one exit, just one way to report an error.
Not sure, need to look closer, but maybe if can at the end report a warning "saving succeded, but backup failed"

Reply:
Hmm, it returns `true` or `false`, and on in the `false` case the caller emits an error message with the string this method filled into `MScore::lastError`. Looks like we'd need a 3rd state.
Or rely on `MScore::lastError` to be empty when everything went fine and in file.cpp after the call to `saveFile()` and chening for success, check whether `MScore::lastError` (or a new `MScore::lastWarning` maybe?) is set and then put out a `QMessage::warning(mscore, tr("Save File"), MScore::lastError);`?

Reply:
This might be out of score for this PR though, it is a problem that (potentially) existed before, here now just with one added possible error, the failing `mkdir()`.
It might help though finding out more about those cases where entirley or largly empty files got saved or something else in the save process went wrong.

Reply:
So I suggest that this PR doesn't take care of the error-checking problems already existing. Not this, nor the problem of rename you brought up in your last review. This PR purely changes the place where the backup files genereate, and another PR (commit) is needed for a thorough overhaul of the error-checking system ;-)

Reply:
Agreed